oaknational/Oak-Web-Application
# fix: useLocalStorage hook should set local storage with initial value

## Description

- Previously this hook (adapted from ts-hooks) wasn't setting the initial value in local storage (it was more of a default value than an initial value)
- This PR ensures that the initial value is set in local storage if no local storage value is present
- adds test for above behaviour


## How to test

1. Go to {cloud link}
2. Check the oak-anonymous-id in local storage
3. Refresh the page and/or open a new tab
4. oak-anonymous-id should be same value


## Checklist
